<launch>

  <arg name="map_frame_id" default="map"/>
  <arg name="map_size_x" default="80"/>
  <arg name="map_size_y" default="80"/>
  <arg name="map_size_z" default="15"/>
  <arg name="map_x_origin" default="-40.0"/>
  <arg name="map_y_origin" default="-40.0"/>
  <arg name="map_z_origin" default="-0.1"/>



  <node pkg ="param_env" name ="read_grid_map" type ="read_grid_map" output = "screen">    
    <!-- this is the map init position, not the odom-->

    <param name="map/x_size"     value="$(arg map_size_x)" />
    <param name="map/y_size"     value="$(arg map_size_y)" />
    <param name="map/z_size"     value="$(arg map_size_z)" />
    <param name="map/x_origin"   value="$(arg map_x_origin)"/>
    <param name="map/y_origin"   value="$(arg map_y_origin)"/>
    <param name="map/z_origin"   value="$(arg map_z_origin)"/>

    <param name="map/resolution"    value="0.1"/>        
    <param name="map/frame_id"      value="$(arg map_frame_id)" />
    <param name="map/inflate_ratio" value="0.3"/>

     <!-- map mode 
           0 randomly generate
           1 read the ros bag poind cloud 1
           2 read the ros bag poind cloud 2
           3 read pcd file -->
    <param name="map/mode"  value="2" />

    <!-- mode 1 eg -->
    <!-- <param name="file_path" value="$(find param_env)/data/pc1/levine/levine.bag"/>
    <param name="bag_topic" value="/cloud"/> -->

    <!-- mode 2 eg -->
    <!-- <param name="file_path" value="$(find param_env)/data/pc2/iss_obs.bag"/>
    <param name="bag_topic" value="/obs_points"/>  -->
    <!-- point clouds topic name in the bag file -->

    <!-- mode 3 eg -->
    <param name="file_path" value="/home/wyw/Code/AirSim_Blocks_Modified_Linux_Build/nn_corridor_generator/datasets/multi_map_dataset/forest_data/"/>



  </node>


  <!--<arg name="rviz_config" default="$(find param_env)/launch/default.rviz"/>
  <node pkg="rviz"
    type="rviz"
    name="rviz"
    args="-d $(arg rviz_config)"/> -->


</launch>